modules/ROOT/pages/keycloak/authenticating.adoc

```diff
@@ -270,6 +270,73 @@ var result = await authService.Logout(authService.CurrentUser());
 
 NOTE: To perform *backchannel or federated logouts*, you must enable the Backchannel Logout option for the federated identity provider. More information is available in the Keycloak documentation under link:https://www.keycloak.org/docs/3.3/server_admin/topics/identity-broker/oidc.html[OICD Identity Providers, window="_blank"].
 
+== Authenticating with a back end service
+
+In a typical scenario, you not only want the user to authenticate on the mobile app, you also want that user to authenticate with a back end service. To achieve this, you must add an additional token:
+
+[source]
+----
+Authorization: BEARER Yourtoken
+----
+
+SDK also provides number of helpers to build authentication headers that will provide way to refresh it once token is outdated.
+
+[role="primary"]
+.Android
+****
+. Get the `AuthHeaderProvider` instance provided by the Aerogear SDK:
++
+[source,java]
+----
+authService.getAuthHeaderProvider()
+----
+. Plug it into your networking layer. When using the `okhttp` library, this can be added to the list of request interceptors to automatically supply the required tokens:
++
+[source,java]
+----
+MobileCore.getInstance()
+          .getHttpLayer()
+          .requestHeaderInterceptor()
+          .add(authService.getAuthHeaderProvider());
+----
+****
+
+[role="secondary"]
+.iOS
+****
+. Get the `AuthHeaderProvider` instance provided by the Aerogear SDK:
++
+[source,swift]
+----
+authService.getAuthHeaderProvider()
+----
+. Plug it into your networking layer.
+****
+
+[role="secondary"]
+.Cordova
+****
+Cordova SDK doesn't provide an HTTP layer so, it also doesn't provide an Auth Header. It's your responsibility to attach that to your HTTP provider.
```

Comment:
1. bit ambiguous here, do you mean:
The AeroGear Cordova SDK does not provide ...
or 
The Apache Cordova SDK does not provide ...

2. therefore it cannot provide an Auth Header.

3. You must attach an HTTP layer to the HTTP provider .

Is that demonstrated in the code below? if not, do we need to give them a link?

Reply:
1. I mean AeroGear Cordova SDK

3. It depends how HTTP provides you are using but is there is an example after this text

@ziccardi Can you update it with the @finp suggestions?

Reply:
@danielpassos @finp : rewrote. Mind giving a look?